x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -295,6 +318,23 @@ private void buildThenMaybeCacheRole(RoleKey roleKey, Collection<RoleDescriptor>
         }, listener::onFailure));
     }
 
+    private void getOrBuildRoleForApiKey(Authentication authentication, boolean limitedBy, ActionListener<Role> roleActionListener) {
+        final Tuple<String, BytesReference> apiKeyIdAndBytes = apiKeyService.getApiKeyIdAndRoleBytes(authentication, limitedBy);
+        final MessageDigest digest = MessageDigests.sha256();
+        digest.update(BytesReference.toBytes(apiKeyIdAndBytes.v2()));
+        final String roleDescriptorsHash = MessageDigests.toHexString(digest.digest());
```

Comment:
Nit: you can can compute the digest in one line.

Reply:
Even more succint:
```
final String roleDescriptorsHash = MessageDigests.toHexString(MessageDigests.sha256().digest(BytesReference.toBytes(apiKeyIdAndBytes.v2())));
```

Reply:
Updated as suggested. Thanks!